### DISCUSSION TITLE
Adding suggestions to inline comments?

### DISCUSSION THREAD

**Discussion by BrunnerLivio** [danger/danger-js#1193] [Ideas]
I'd like to automatically give suggestions in inline comments. If I am mistaken this is currently not possible with Danger?

From my knowledge, both Github and Gitlab support the following in inline comments:

*Github*:
````markdown
```suggestion
Here is my suggestion
```
````

*Gitlab*:

````markdown
```suggestion:-0+0
+Here is my suggestion
```
````

Example in Gitlab:

![Screenshot 2022-01-10 at 10 30 02](https://user-images.githubusercontent.com/9899423/148743838-e469191d-7abb-473d-bb8f-fcbef6cbcde6.png)

Maybe the following could be considered:

```typescript
// Note: suggestion parameter has been added
warn(message: MarkdownString, file?: string, line?: number, suggestion?: string) => void
```

**Comment by fbartho** [comment]
If you embed your suggestion in the `message: MarkdownString` it will be attached to the provided file/line.

So I don’t believe we need this extra API argument.

**Reply by orta** [reply]
Yeah, agree

**Reply by BrunnerLivio** [reply]
Thanks a lot that worked.

````typescript
warn('My message\n\n```suggestion:-0+0\nSuggestion\n```, path, line);
````

A utility function to generate the suggestion string depending on the platform would be nice, but that'd be only-nice-to-have.
